compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java

```diff
@@ -0,0 +1,110 @@
+    private static final String MODULE_NAME = "ModuleName";
+    private static final String VERSION = "Version";
+
+    private static void setFunctionDescriptions(PackageDescriptor descriptor, String functionName, Component component) {
+        String orgName = descriptor.org().value();
+        String moduleName = descriptor.name().value();
+        String version = String.valueOf(descriptor.version().value().major());
+
+        Param functionNameParam = new Param(FUNCTION_NAME, functionName);
+        Param orgParam = new Param(ORG_NAME, orgName);
+        Param moduleParam = new Param(MODULE_NAME, moduleName);
+        Param versionParam = new Param(VERSION, version);
+
+        component.setParam(functionNameParam);
+        component.setParam(orgParam);
+        component.setParam(moduleParam);
+        component.setParam(versionParam);
+    }
+
+    private static void setArguments(FunctionSymbol functionSymbol, Component component) {
+        int noOfParams = 0;
+        Optional<List<ParameterSymbol>> params = functionSymbol.typeDescriptor().params();
+        if (params.isPresent()) {
+            noOfParams = params.get().size();
+        }
+
+        for (int i = 0; i < noOfParams; i++) {
+            if (params.get().get(i).getName().isEmpty()) continue;
+            Param param = new Param(Integer.toString(i), functionSymbol.typeDescriptor().params().get().get(i).getName().get());
+            component.setParam(param);
+        }
+        Param sizeParam = new Param(SIZE, Integer.toString(noOfParams));
+        component.setParam(sizeParam);
+    }
+
+    private static void setIcon(SyntaxNodeAnalysisContext context, Connector connector) {
+        Optional<BallerinaToml> ballerinaToml = context.currentPackage().ballerinaToml();
+        if (ballerinaToml.isEmpty()) return;
+
+        TomlDocument tomlDocument = ballerinaToml.get().tomlDocument();
+        TomlTableNode path = (TomlTableNode) tomlDocument.toml().rootNode().entries().get(Connector.TOML_ICON_NODE);
+        if (path == null) return;
+        TomlKeyValueNode iconPath = (TomlKeyValueNode) path.entries().get(Connector.TOML_ICON_KEY);
+
+        if (iconPath == null) return;
+        TomlStringValueNode valuePath = (TomlStringValueNode) iconPath.value();
+        connector.setIconPath(valuePath.getValue());
+    }
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+
+        SemanticModel semanticModel = context.compilation().getSemanticModel(context.moduleId());
+
+        if (!(context.node() instanceof AnnotationNode annotationNode)) return;
+        Optional<Symbol> symbol = semanticModel.symbol(annotationNode);
+        if (symbol.isEmpty()) return;
+        if (!(symbol.get() instanceof AnnotationSymbol annotationSymbol)) return;
+        Optional<String> annotationName = annotationSymbol.getName();
+        if (annotationName.isEmpty()) return;
+        if (!annotationName.get().equals(Component.ANNOTATION_QUALIFIER)) return;
```

Comment:
Shall we move this validation into a separate method?

Reply:
moved only the annotation node part, do we have to move the function part it takes lot of return values